src/components/Dialog/__tests__/Dialog.interactions.ts

```diff
@@ -0,0 +1,59 @@
+import { resetFocus } from "../../../__tests__/interactions-helper";
+import { expect } from "@storybook/jest";
+import { BoundFunctions, Screen } from "@testing-library/react";
+import {
+  CONTEXT_MENU_DIALOG,
+  CLICK_OUTSIDE_DIALOG,
+  HIDE_TRIGGERS_CONTAINER,
+  CLICK_OUTSIDE_DIALOG_BUTTON
+} from "./DialogDataTestIds";
+import { getByTestId, interactionSuite } from "../../../tests/interactions-utils";
+import { queries, userEvent, fireEvent, waitFor } from "@storybook/testing-library";
+
+const isDialogHiddenAfterClickOutside = createTestIfDialogHiddenAfterTrigger(CLICK_OUTSIDE_DIALOG, () =>
+  userEvent.click(getDialogContainer())
+);
+const isDialogHiddenAfterContextMenu = createTestIfDialogHiddenAfterTrigger(CONTEXT_MENU_DIALOG, () =>
+  fireEvent.contextMenu(getDialogContainer())
+);
+
+export const closeTriggersInteractionSuite = interactionSuite({
+  tests: [isDialogHiddenAfterClickOutside, isDialogHiddenAfterContextMenu],
+  beforeAll: async canvas => {
+    // From some reason we have an issue with rendering the dialogs according to the container selector in the initial mount, after clicking
+    // all dialog render in the right placements
+    const clickOutsideButton = await getByTestId(canvas, CLICK_OUTSIDE_DIALOG_BUTTON);
+    await userEvent.click(clickOutsideButton);
+    // wait for hide default delay
+    await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
We can also use the hide delay and show delay props in the story but the stories are too big anyway.
I guess you may think it is a signal for that we need to separate between the stories and the tests in the future, there is a chance that is the case, but also the dialog api is really complex compare to other components in Vibe.